### PR TITLE
Updating ExpectedException tests.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
* Updating with comments.
* Removing StringBuilder wherever used.
* Use Asserts correctly.

### CODE REVIEW

File: src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.0.0.cs [hunk 1]
```diff
@@ -24,30 +19,34 @@ public partial class ExpectedExceptionTests : ConditionalWcfTest
...
-            if (e.Message != exceptionMsg)
-            {
-                Assert.True(false, string.Format("Expected Fault Message: {0}, actual: {1}", exceptionMsg, e.Message));
-            }
-            return;
-        }
-
-        Assert.True(false, "Expected ActionNotSupportedException exception, but no exception thrown.");
+        // *** ADDITIONAL VALIDATION *** \\
+        Assert.Equal(exception.Message, exceptionMsg);
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.0.0.cs]: Nit: I'd probably make this an Assert.True(String.Equals(), String.Format("Expected exception message {0} but actual was {1}... just to help diagnose in a lab failure.

File: src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.0.0.cs [hunk 2]
```diff
@@ -57,142 +56,158 @@ public static void NonExistentAction_Throws_ActionNotSupportedException()
...
+                // *** ENSURE CLEANUP *** \\
+                watch.Stop();
+                ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
+            }
+        });
 
+        // *** ADDITIONAL VALIDATION *** \\
         // want to assert that this completed in > 5 s as an upper bound since the SendTimeout is 5 sec
         // (usual case is around 5001-5005 ms) 
         Assert.InRange<long>(watch.ElapsedMilliseconds, 4985, 6000);
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.0.0.cs]: >Assert.InRange [](start = 8, length = 14)

Nit: I might make this an Assert.True so we could have a more descriptive error message to help diagnose a lab failure.  Not blocking.

File: src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.1.0.cs [hunk 3]
```diff
@@ -26,19 +18,35 @@ public partial class ExpectedExceptionTests : ConditionalWcfTest
...
+            finally
+            {
+                // *** ENSURE CLEANUP *** \\
+                ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
             }
         });
 
+        // *** ADDITIONAL VALIDATION FOR NET NATIVE *** \\
         // On .Net Native retail, exception message is stripped to include only parameter
         Assert.True(exception.Message.Contains(nonExistentHost), string.Format("Expected exception message to contain: '{0}'", nonExistentHost));
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.1.0.cs]: >"Expected exception message to contain: '{0}' [](start = 79, length = 45)

Nit: Not blocking. I know this code existed before your changes, but this message would be improved if it said what Exception.Message was, not just what it did not contain.

**Comment by roncain** [comment]
>                 Assert.True(false, string.Format("Expected type SecurityNegotiationException, Actual: {0}", exceptionType));

Nit: not blocking.  This Assert would be improved with a full exception.ToString().  If we get the wrong exception, we want to know everything it can tell us.

---
Refers to: src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.1.0.cs:335 in 455d1b0. [](commit_id = 455d1b06ddba2ba6e656799e618f1123b05c5073, deletion_comment = False)

**Comment by StephenBonikowsky** [comment]
Updated with nits, this is the right time to make all needed changes.

**Pull request merged**